src/main/level_save_info.cpp

```diff
@@ -0,0 +1,255 @@
+/*
+ * TheXTech - A platform game engine ported from old source code for VB6
+ *
+ * Copyright (c) 2009-2011 Andrew Spinks, original VB6 code
+ * Copyright (c) 2020-2023 Vitaly Novichkov <admin@wohlnet.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdio>
+#include <string>
+#include <bitset>
+#include <PGE_File_Formats/file_formats.h>
+#include <Logger/logger.h>
+#include <fmt_format_ne.h>
+
+#include "npc_id.h"
+
+#include "globals.h"
+
+static bool s_serializeSaveInfo(std::string& s, const LevelSaveInfo_t& info)
+{
+    if(!info.inited())
+        return false;
+
+    char medals_got[9];
+    char medals_best[9];
+
+    for(int i = 0; i < info.max_medals && i < 8; ++i)
+    {
+        medals_got[i]  = (info.medals_got  & (1 << i)) ? '+' : '.';
+        medals_best[i] = (info.medals_best & (1 << i)) ? '+' : '.';
+    }
+
+    for(int i = info.max_medals; i < 8; ++i)
+    {
+        medals_got[i] = ' ';
+        medals_best[i] = ' ';
+    }
+
+    medals_got[8] = '\0';
+    medals_best[8] = '\0';
+
+    unsigned max_stars = info.max_stars;
+    unsigned max_medals = info.max_medals;
+
+    if(max_medals > 8)
+        max_medals = 8;
+
+    s = fmt::sprintf_ne("V:1;Sm:%3u;Mm:%1u;Mg:%s;Mb:%s;", (unsigned)max_stars, (unsigned)max_medals, medals_got, medals_best);
+
+    return true;
+}
+
+static bool s_parseSaveInfo(LevelSaveInfo_t& info, const std::string& s)
+{
+    info = LevelSaveInfo_t();
+
+    unsigned version = 0;
+    int end_pos = 0;
+
+    if(sscanf(s.c_str(), "V:%2u;%n", &version, &end_pos) != 1 || end_pos < 1)
```

Comment:
Please use `SDL_sscanf`

Reply:
Ah, good note, thanks. I generally forget which functions have SDL versions. But, I'll delete this whole file when I update it to use PGE-FL.

Reply:
Little note: recently I already added new stuff into PGE-FL, please take a look.